arch/src/filter_context.rs

```diff
@@ -282,7 +282,10 @@ impl RootContext for FilterContext {
         // No StreamContext can be created until the Embedding Store is fully initialized.
         let embedding_store = match self.mode {
             GatewayMode::Llm => None,
-            GatewayMode::Prompt => Some(Rc::clone(self.embeddings_store.as_ref().unwrap())),
+            GatewayMode::Prompt => match self.embeddings_store.as_ref() {
+                None => return None,
```

Comment:
```suggestion
                None => None,
```

Reply:
I did intend to `return None`. The whole point here is that a StreamContext should not be created if the Embedding Store is not present

Reply:
ah I see it now! make sense.